qa/rpc-tests/test_framework/comptool.py

```diff
@@ -247,24 +247,39 @@ def transaction_requested():
     # Verify that the tip of each connection all agree with each other, and
     # with the expected outcome (if given)
     def check_results(self, blockhash, outcome):
+        waitLoop = 0
+        while 1:  # sometimes connection 0 hasn't figured itself out yet (but the network thread will handle this async)
+            cxn0bbh = self.connections[0].cb.bestblockhash
+            if cxn0bbh != None: break
+            time.sleep(.1)
+            waitLoop+=1
+            if waitLoop > 50:
+                raise AssertionError("Test malfunction -- connection 0 is has no best block for comparision")
         with mininode_lock:
+            if blockhash is None:
+                print("Blockhash is None in check_results")
+                raise AssertionError("Test failed -- passed check_results None as blockhash, was expecting %s" % str(outcome))
+            if cxn0bbh is None:
+                print("connection 0 bestblockhash is None in check_results")
```

Comment:
Are those `print`s statements a debug temporary aid or something we actually need all the time?

Reply:
you are right they aren't really needed now that I'm raising an Assertion in these cases